### ISSUE TITLE
Fails to compile on Arch Linux

### ISSUE THREAD

**Issue by gatesphere** [polluxsynth/xtor#6]
Running a 'make' on Arch Linux gives the following:

```
$ make
gcc  -ansi -Werror -c -o xtor.o xtor.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o dialog.o dialog.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o blofeld_ui.o blofeld_ui.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o blofeld_params.o blofeld_params.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o knob_mapper.o knob_mapper.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o blofeld_knobs.o blofeld_knobs.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o nocturn.o nocturn.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o beatstep.o beatstep.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
gcc  -ansi -Werror -c -o midi.o midi.c `pkg-config --cflags libglade-2.0 gmodule-2.0 alsa` -DUI_DIR=\".\" -g -O2
In file included from /usr/include/alsa/asoundlib.h:54:0,
                 from midi.c:23:
/usr/include/alsa/pcm.h:1012:15: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
 static inline void snd_pcm_pack_audio_tstamp_config(unsigned int *data,
               ^~~~
/usr/include/alsa/pcm.h:1020:15: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
 static inline void snd_pcm_unpack_audio_tstamp_report(unsigned int data, unsigned int accuracy,
               ^~~~
make: *** [Makefile:51: midi.o] Error 1
```

Was able to get it to compile by swapping out `-ansi` for `-std=c99` in the Makefile.  Seems to work fine.

**Comment by polluxsynth** [comment]
Thanks for the heads up. When I swap out -ansi for -std=c99 on my (getting old) Debian Jessie, it doesn't compile due to a bunch of warnings turned into errors thanks to -Werror though. Did you need to do that to get it to compile too?

They are not false positives either, so I need to take a closer look at this. I remember the -ansi came from a Makefile I was originally using from another project, so I could really just take that out, however, that causes other errors, so I need to look over the source and clean this up.

**Comment by gatesphere** [comment]
Hmm, it appears I did take those out while trying to get it to compile.  I guess I forgot to put it back in, once I got it to compile I was happy.  Maybe I should have been more cautious.

**Comment by polluxsynth** [comment]
Well, looking at the resulting warnings with -std=c99 and no -Werror does show that the code has some issues (mainly return statements in non-void functions), but those issues are still there when compiled with -ansi, so when it comes to how the code actually works it shouldn't make a practical difference. Looking in particular at those return statements, they are error exits out of functions when something really bad has happened, so it's not something that would be hit in normal usage anyway.

So, bottom line, if it works for you without -std=c99 and no -Werror go with it, as it should be no worse than in my builds, even though there are formal warnings. But I don't like to have warnings in the code and in rare cases especially hitting one of those incorrect return statements could cause the application to behave strangely, and they should be easy to iron out, so I'll see if I can have a closer look at it over the coming days.

**Comment by polluxsynth** [comment]
BTW, I also tried compiling even without -std=c99, as I usually don't specify any particular C standard when I write code; this however caused the build to completely fail as 'dprintf' apparently is present in stdio.h under certain circumstances. That's the case I should really fix.

**Comment by polluxsynth** [comment]
I've fixed the code now so that it compiles without -ansi or -std=c99, as that should be what is the most portable, and updated the Makefile accordingly. Also fixed a mildly annoying bug regarding the Device Number parameter on the last page, which a) is blank when xtor is started up, and b) was actually not used when sending parameter updates, only when requesting/sending while patches as a bulk dump.

Could you give it a try, on the dev branch, and see if it works for you too, please?

**Comment by gatesphere** [comment]
Dev branch works beautifully for me now.  Thanks!

**Comment by gatesphere** [comment]
Well, almost.  I had to comment out the ``UI_DIR = .``` in the Makefile to get ```make install``` to install a working copy.  Those UI files are important! :)

**Comment by polluxsynth** [comment]
I've fixed the code now so that it compiles without -ansi or -std=c99, as that should be what is the most portable, and updated the Makefile accordingly. Also fixed a mildly annoying bug regarding the Device Number parameter on the last page, which a) is blank when xtor is started up, and b) was actually not used when sending parameter updates, only when requesting/sending while patches as a bulk dump.

Could you give it a try, on the dev branch, and see if it works for you too, please?

**Comment by polluxsynth** [comment]
Thanks for testing!

Yeah, the UI_DIR = . -line should really be commented out on at least on the master branch, although it looks like I've missed that detail for quite some time now. I'll try to put together a proper release within a couple of days.

**Comment by polluxsynth** [comment]
There, now the master branch should build out of the box, with the version number set to 1.5 .
